main.go

```diff
@@ -126,20 +126,18 @@ func main() {
 	customSetupHealthChecks(mgr)
 	customSetupEndpoints(pprofActive, mgr)
 
+	options := controllers.SetupOptions{
+		SupportExtendedDaemonset: supportExtendedDaemonset,
+		APIKey:                   os.Getenv(config.DDAPIKeyEnvVar),
```

Comment:
Would there be a case for supporting multiple sets of creds in the future? Example that comes to mind - monitors created by different teams operating on the same cluster. We currently support one set of creds per DatadogAgent resource, this difference with monitors might be confusing. At the same time DatadogAgent is usually a singleton, we don't have a good multi-tenancy story for it.

Reply:
i **think** i discussed this with someone (vincent?) before, and it was decided to have a separate controller per namespace that would be watched independently and could have different credentials. i think this would cover most cases of teams operating on the same cluster; what do you think?

Reply:
That makes a lot of sense, thank you.